syntax = "proto3";
import "google/protobuf/struct.proto";

service EdaGameService{
    rpc CreateGame(CreateGameRequest) returns(CreateGameResponse){}
    rpc ExecuteAction(ExecuteActionRequest) returns(GameStateResponse){}
    rpc EndGame(EndGameRequest) returns(GameStateResponse) {}
    rpc Penalize(PenalizeRequest) returns(GameStateResponse) {}
}

message CreateGameRequest {
    repeated string players = 1;
}

message CreateGameResponse {
    string idgame = 1;
}

message ExecuteActionRequest {
    string idgame = 1;
    google.protobuf.Struct data= 2;
}

message EndGameRequest {
    string idgame = 1;
}

message PenalizeRequest {
    string idgame = 1;
}

message GameStateResponse {
    map <string,google.protobuf.Struct> play_data = 1;
    string current_player = 2;
    string next_player = 3;
    google.protobuf.Struct turn_data = 4;
}

//python -m grpc_tools.protoc -I. --python_out=. --grpc_python_out=. ./eda_games.proto

